### DISCUSSION TITLE
Possible to return foreign table after insert()?

### DISCUSSION THREAD

**Discussion by jimmiejackson414** [supabase/supabase#7942] [Questions] [answered]
I know that by default, an insert returns the affected rows. Is it possible to return data from a foreign table?

I'm looking to do something like this:

```typescript
const { data, error } = await supabase
  .from('items')
  .insert(formattedData)
  .select('*, category_items(*, category(*))');
console.log({ data });
console.log({ error });
```

**Comment by GaryAustin1** [comment, marked as answer]
So it appears PostgREST can handle it: https://postgrest.org/en/stable/api.html#embedding-after-insertions-updates-deletions
The question is how to make it happen from supabase-js.  You could always issue the fetch call directly if not.  
Edit: I assume what you show did not work?

**Reply by GaryAustin1** [reply]
```
let response = await supabase
        .from('messages')
        .insert([{message:'hi there',test5_link:6}])
        .select('*,test5(*)')
```
![image](https://user-images.githubusercontent.com/54564956/181655301-c3ff6830-3df9-4bf4-8e56-19f0a7d246ba.png)


And it works!  Cool.

**Reply by jimmiejackson414** [reply]
@GaryAustin1 wow that worked for you?? I must have a relationship set up incorrectly or something, because it's not working for me at all and I'm not getting any errors. I'll review the relationships, but it's great to know that it at least worked in your test. Thanks!

**Reply by GaryAustin1** [reply]
RLS? Also try without the deeper link (although I don't see why that would matter).
The network url looked something like this for me:
```
https://ugcxxxxxxxxbseypbq.supabase.co/rest/v1/messages?columns="message","test5_link"&select=*,test5(*)
```
Of course with escapes for the quotes, etc.
Sorry for the double post, don't like putting my test site url out.

**Reply by jimmiejackson414** [reply]
Perfect, this will be very handy to reference. Thanks again!
